src/irclib.py

```diff
@@ -957,6 +979,11 @@ def do354(self, irc, msg):
         (n, t, user, ip, host, nick, status, account, gecos) = msg.args
         hostmask = '%s!%s@%s' % (nick, user, host)
         self.nicksToHostmasks[nick] = hostmask
+        if account == '0':
+            # logged out
+            self.nicksToAccounts[nick] = None
```

Comment:
Storing None values seems a bit odd, would it be better to delete the entry?

Reply:
no entry means "unknown", `None` entry means "logged out"

Reply:
It might be nice to have the function describe how it'll throw KeyError if the user isn't in the state